mistralrs-core/src/utils/gguf_metadata.rs

```diff
@@ -0,0 +1,93 @@
+use akin::akin;
+use anyhow::ensure;
+use anyhow::Result;
+use candle_core::quantized::gguf_file;
+use std::collections::HashMap;
+
+pub struct MetadataContext<'a> {
```

Comment:
> Can you please extend its usage to the `quantized_*.rs` in `models` and `xlora_models`?

Yes, I can tackle that in about 12 hours.

Just need guidance on the current test failure.

---

> we can incorporate some subtyping support for backwards compatibility
> (for example for GGUF, u64 fields were previously documented as being u32. A PR of mine was merged into Candle).

I'm aware of the PR I followed that one. It might be a bit awkward to get similar support here.

I match a rust primitive type with the expected `Value` method to get that type. If the wrapped value is not valid, the `Value` method will return an error. Working around that might be more hassle than improving the `Value` enum upstream?

Reply:
> Just need guidance on the current test failure.

I'll take a look. 

> I match a rust primitive type with the expected Value method to get that type. If the wrapped value is not valid, the Value method will return an error. Working around that might be more hassle than improving the Value enum upstream?

The way I did it in the PR was to match all possible subtypes in addition to the target type. However, we probably actually do not need to do this as I cannot think of any use cases other than u64.